tools/devops/device-tests/README.md

```diff
@@ -73,6 +73,66 @@ The are two main groups of variables used in the pipelines:
 - common (under templates/common-variables.yml): Contains a set of group variables that have been created by release engineering and that contain all the different secrets needed by the pipeline.
 - lab (under templates/[ddfun|cambridge]-variables: Contains the different configurations per lab. That includes the pools to use and the type of storage.
 
+There are certain environment variables that are created by steps in the first job that are later shared with the second job. In order to do that you
+have to ensure that you do the following:
+
+```yaml
+- bash: |
+    set -x
+    set -e
+    echo 'Use xamarin storage $USE_XAMARIN_STORAGE'
+
+    cd $WORKING_DIR/xamarin-macios
+    if [[ "$USE_XAMARIN_STORAGE" == "True" ]]; then
+      EC=0
+      ssh builder@xamarin-storage "mkdir -p /volume1/storage/$XAMARIN_STORAGE_PATH" || EC=$?
+      if [ $EC -eq 0 ]; then
+        echo '##vso[task.setvariable variable=XAMARIN_STORAGE_FAILED;isOutput=true]false'
+        echo '##vso[task.setvariable variable=XAMARIN_STORAGE_PATH;isOutput=true]$XAMARIN_STORAGE_PATH'
+        export TESTS_PERIODIC_COMMAND="--periodic-interval 10 --periodic-command rsync --periodic-command-arguments '-avz -e \"ssh\" $PWD/jenkins-results builder@xamarin-storage:/volume1/storage/$XAMARIN_STORAGE_PATH'"
+      else
+        echo '##vso[task.setvariable variable=XAMARIN_STORAGE_FAILED;isOutput=true]true'
+        echo '##vso[task.setvariable variable=XAMARIN_STORAGE_PATH;isOutput=true]""'
+      fi
+    fi
+
+    make -C builds download .stamp-mono-ios-sdk-destdir -j
+    MONO_ENV_OPTIONS=--trace=E:all make -C tests vsts-device-tests
+  env:
+    WORKING_DIR: $(System.DefaultWorkingDirectory) 
+    XAMARIN_STORAGE_PATH: $(XAMARIN_STORAGE_PATH)
+    TESTS_EXTRA_ARGUMENTS: ${{ parameters.testsLabels }}
+    USE_XAMARIN_STORAGE: '${{ parameters.useXamarinStorage }}'
+    VSDROPS_URI: '${{ parameters.vsdropsPrefix }}/$(Build.BuildNumber)/$(Build.BuildId);/' # uri used to create the vsdrops index using full uri
+  displayName: 'Run tests'
+  name: runTests 
+  timeoutInMinutes: 600
+```
+
+There are two details we have to pay attention here, first:
+
+```yaml
+name: runTests 
+```
+
+We need to set the name of the step. This should not be confused with the displayName which is simply a string used in the vsts webpage.
+Later we can access the output variables of the step in the following way:
+
+```yaml
+
+XAMARIN_STORAGE_PATH: $[ dependencies.tests.outputs['runTests.XAMARIN_STORAGE_PATH'] ]
+```
+
+The other piece of code is
+
+```yaml
+
+echo '##vso[task.setvariable variable=XAMARIN_STORAGE_FAILED;isOutput=true]true'
+echo '##vso[task.setvariable variable=XAMARIN_STORAGE_PATH;isOutput=true]""'
+```
+
+As you can see, we use **isOutput=true** which means that the environment variable will be registered as an output variable of the step.
```

Comment:
Not sure where it would be best to put, but I'm wondering whether you could add a small note about making sure that vars required by VSDrops are accessible across machines?